### PR TITLE
Ex02

### PR DESCRIPTION
**Pull request by mikkonator97**
Note that file_stats.py will raise a value error if you have characters outside of the UTF-8, which i added in order to practice on raising errors. If that's a problem it is documented how to easy make it work anyways.

### CODE REVIEW

File: src/mikko_rekstad_ex/ex02/bubble_sort.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    n = len(data2)
+    swapped = True
+    # Using a while loop that runs until the list is sorted.
+    while swapped:
+        # The list will be sorted when  swapped remains False.
+        swapped = False
+        for i in range(n-1):
+            if data2[i] > data2[i+1]:
+                # Using the swap function to swap places in  the list.
+                data2 = swap(list(data2), i+1, i)
```
Comment (huynhngoc) [inline src/mikko_rekstad_ex/ex02/bubble_sort.py]: Why make it too complicated here, 
`data2[i], data2[i+1] = data2[i+1], data2[i]` can easily work

File: src/mikko_rekstad_ex/ex02/bubble_sort.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+def bubble_sort(data2):
+    """Input is the list we want to sort.
+    ----
+    Output will be the sorted list."""
+    # Defining variables.
+    n = len(data2)
+    swapped = True
+    # Using a while loop that runs until the list is sorted.
+    while swapped:
```
Comment (huynhngoc) [inline src/mikko_rekstad_ex/ex02/bubble_sort.py]: Works, but may cause confusion when walk-through. And will make some redundant comparison of sorted data. You may want to check the suggested solution for this exercise.

File: src/mikko_rekstad_ex/ex02/file_stats.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+
+
+def char_counts(textfilename):
+    """Takes input filename.
+    ---
+    Output will be the dictionary contaning the counted amount of each
+    character in the file belonging to the input filename."""
+    text = open(textfilename)
```
Comment (huynhngoc) [inline src/mikko_rekstad_ex/ex02/file_stats.py]: Use `with` statement when opening file would be better.

File: src/mikko_rekstad_ex/ex02/file_stats.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    # Defining the dictionary with 0 for each character.
+    dictionary = {k: 0 for k in range(256)}
```
Comment (huynhngoc) [inline src/mikko_rekstad_ex/ex02/file_stats.py]: It won't have to be this complicated. You may also want to check on the suggested solution for this task

Comment (mikkonator97) [reply src/mikko_rekstad_ex/ex02/file_stats.py]: Will do, thanks for feedback ! :D

**Pull request merged**